translator class from raw state vector / lcm message for inverse dynamics

siyuanfeng-tri opened the pull request:
refactored previous lcm message -> humanoid status translator to a general base class that outputs humanoid status, and 2 derived class for raw state vector input and lcm message input respectively.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/5295)
<!-- Reviewable:end -->

siyuanfeng-tri commented:
+@liangfok for feature review.
part 3 of generalizing the inverse dynamics controller to work with iiwa. thanks!

---

Review status: 0 of 8 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-:-Kdh2dqRgjj1bMiQSui7:b-kgzuh7)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Completed first pass.

---

Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.cc, line 98 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxG9ayuXXiv0ckvA_x:-KdxG9ayuXXiv0ckvA_y:b-jbjm9o) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.cc#L98)):*
> ```cpp
> }
> 
> void HumanoidStatus::Update(
> ```

BTW, consider having this method call the method above.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 105 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxOuAMzTsDl6RMC6p8:-KdxOuANgxD4Q1_EH_nN:bzdqpjz) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L105)):*
> ```objc
>   /**
>    * Do kinematics and compute useful information based on kinematics and
>    * measured force torque information.
> ```

BTW, consider documenting the fact at this method simply saves the values of the various parameters and calls `Update()`.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 115 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxO5a4wKfGirmXI7K0:-KdxO5a4wKfGirmXI7K1:b-zb9cb9) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L115)):*
> ```objc
>    * @param r_wrench is wrench measured in the sensor frame.
>    */
>   void Update(double t, const Eigen::Ref<const VectorX<double>>& q,
> ```

This method is overloaded twice. The style guide discourages overloading methods. Consider renaming them so they can be more easily distinguished and understood.

https://google.github.io/styleguide/cppguide.html#Function_Overloading

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-KdwiQvHpyIpmL8pjQXn:bic4in9) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
Consider revising this paragraph to be more consistent with the previous method:

> Do kinematics and compute useful information based on kinematics and an assumption that all force and torque measurements are zero.

Alternatively, you can document that this method calls the method above with `joint_torque`, `l_wrench`, and `r_wrench` all set to zero. This may be easier if the method above were named differently (see comment above regarding method overloading).

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 131 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxNGU9ib4qNl7JqsEi:-KdxNGUA_WeR3kKxL-3G:b4xsxw1) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L131)):*
> ```objc
>               const Eigen::Ref<const VectorX<double>>& v);
> 
>   void Update();
> ```

BTW, since this method does the actual work of the above two methods, I think it should be fully documented.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.cc, line 40 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxS_9f9fim-fyvHdtI:-KdxS_9f9fim-fyvHdtJ:b-5tt6x6) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.cc#L40)):*
> ```cpp
>     const systems::Context<double>& context,
>     systems::SystemOutput<double>* output) const {
>   VectorX<double> x = EvalEigenVectorInput(context, input_port_index_state_);
> ```

BTW, `const`?

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.cc, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxSdFwNzsXG6ApLOmS:-KdxSdFwNzsXG6ApLOmT:bd78kso) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.cc#L42)):*
> ```cpp
>   VectorX<double> x = EvalEigenVectorInput(context, input_port_index_state_);
> 
>   HumanoidStatus& hum_status =
> ```

BTW, consider naming this variable "status" since "hum" is an abbreviation and the style guide discourages abbreviations.

(Same comment below.)

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 16 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwkmcEZvcu41wWCwFr:-KdwkmcEZvcu41wWCwFs:bldms2c) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L16)):*
> ```objc
> 
> /**
>  * Base class that translates different forms of state into HumanoidStatus
> ```

Missing period at end of sentence.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 16 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwnoU5ExvA8RCl1vNE:-KdwnoU5ExvA8RCl1vNF:bjd5wki) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L16)):*
Consider rewording:

> A base class for systems that translate various types of state into HumanoidStatus.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Kdwk8m_gZkWrs46ww_h:-Kdwk8m_gZkWrs46ww_i:b-cx21w1) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L25)):*
BTW, consider replacing:

> An internal alias is saved, so the

with:

> The

Since the fact that an alias is created is an internal implementation detail that need not be exposed to readers.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 52 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwoeH_ConxpChtjPag:-KdwoeH_ConxpChtjPah:b-hoircc) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L52)):*
> ```objc
>  private:
>   const RigidBodyTree<double>& robot_;
>   const std::string kAliasGroupPath;
> ```

This variable cannot use the `kFoo` naming syntax since its value is set based on a constructor input parameter meaning it might change from one instance to the next:

https://google.github.io/styleguide/cppguide.html#Constant_Names

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 58 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwoGKhxZ0qsbpDJfBU:-KdwoGKhxZ0qsbpDJfBV:b-rw4jld) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L58)):*
> ```objc
> 
> /**
>  * Extends HumanoidStatusTranslatorSystem to translates a state vector to
> ```

BTW, consider deleting "Extends HumanoidStatusTranslatorSystem to" since it's implied by the class hierarchy.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 68 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwkYLf0ZC190LFacfS:-KdwkYLf0ZC190LFacfT:b6ggl17) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L68)):*
BTW, consider replacing:

> An internal alias is saved, so the

with:

> The

Since the fact that an alias is created is an internal implementation detail that need not be exposed to readers.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 91 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Kdwo6TNThu6yLYxzD6I:-Kdwo6TNThu6yLYxzD6J:b-rw4jld) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L91)):*
> ```objc
> 
> /**
>  * Extends HumanoidStatusTranslatorSystem to translates a
> ```

BTW, consider deleting "Extends HumanoidStatusTranslatorSystem to" since it's implied by the class hierarchy.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwkiEfsVB2jc8-_UqP:-KdwkiEfsVB2jc8-_UqQ:b6ggl17) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L102)):*
BTW, consider replacing:

> An internal alias is saved, so the

with:

> The

Since the fact that an alias is created is an internal implementation detail that need not be exposed to readers.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-:-KdxTBKvmQ8DeiaGmVK-:bl4rucz)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
Review status: 4 of 8 files reviewed at latest revision, 15 unresolved discussions.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 105 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxOuAMzTsDl6RMC6p8:-Ke-AAUpOY3Qff6XwXCo:b-g1o1bh) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L105)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider documenting the fact at this method simply saves the values of the various parameters and calls `Update()`.
</blockquote></details>

i move Update() to private. 

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 115 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxO5a4wKfGirmXI7K0:-Ke-AEIAjcbswCTTOYvk:bbwsvsv) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L115)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

This method is overloaded twice. The style guide discourages overloading methods. Consider renaming them so they can be more easily distinguished and understood.

https://google.github.io/styleguide/cppguide.html#Function_Overloading
</blockquote></details>

renamed the other public one. 

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-Ke-AMtajKojvlPFJHH_:b-olt9ip) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Consider revising this paragraph to be more consistent with the previous method:

> Do kinematics and compute useful information based on kinematics and an assumption that all force and torque measurements are zero.

Alternatively, you can document that this method calls the method above with `joint_torque`, `l_wrench`, and `r_wrench` all set to zero. This may be easier if the method above were named differently (see comment above regarding method overloading).
</blockquote></details>

renamed this. 

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 131 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxNGU9ib4qNl7JqsEi:-Ke-AM88VMG4PYYdKNdk:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L131)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, since this method does the actual work of the above two methods, I think it should be fully documented.
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.cc, line 40 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxS_9f9fim-fyvHdtI:-Ke-1f-lKSzti04WFs95:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.cc#L40)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, `const`?
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.cc, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxSdFwNzsXG6ApLOmS:-Ke-1ecpdeSjSO4zfroG:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.cc#L42)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider naming this variable "status" since "hum" is an abbreviation and the style guide discourages abbreviations.

(Same comment below.)
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 16 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwkmcEZvcu41wWCwFr:-Ke-1k-b3luFBITmptoj:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L16)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Missing period at end of sentence.
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 16 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwnoU5ExvA8RCl1vNE:-Ke-1pVjBKZfGWH-RILV:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L16)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Consider rewording:

> A base class for systems that translate various types of state into HumanoidStatus.
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Kdwk8m_gZkWrs46ww_h:-Ke-1vIWzKV9NObgrKT_:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L25)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing:

> An internal alias is saved, so the

with:

> The

Since the fact that an alias is created is an internal implementation detail that need not be exposed to readers.

</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 52 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwoeH_ConxpChtjPag:-Ke-22ZhmPTSZ_00HplI:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L52)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

This variable cannot use the `kFoo` naming syntax since its value is set based on a constructor input parameter meaning it might change from one instance to the next:

https://google.github.io/styleguide/cppguide.html#Constant_Names
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 58 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwoGKhxZ0qsbpDJfBU:-Ke-2AfAZOYezH5lwTCj:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L58)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider deleting "Extends HumanoidStatusTranslatorSystem to" since it's implied by the class hierarchy.
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 68 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwkYLf0ZC190LFacfS:-Ke-2ItylcsbPbsfFzj_:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L68)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing:

> An internal alias is saved, so the

with:

> The

Since the fact that an alias is created is an internal implementation detail that need not be exposed to readers.
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 91 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Kdwo6TNThu6yLYxzD6I:-Ke-2JhYQK2dKIuiMPcw:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L91)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider deleting "Extends HumanoidStatusTranslatorSystem to" since it's implied by the class hierarchy.
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwkiEfsVB2jc8-_UqP:-Ke-2Kgjr6J6XIdwxE5E:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L102)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing:

> An internal alias is saved, so the

with:

> The

Since the fact that an alias is created is an internal implementation detail that need not be exposed to readers.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-Ke0JFgmvIQqi11x8Qr1:bj2ijsr) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

renamed this. 
</blockquote></details>

BTW, "everything related" is perhaps too informal and vague. Consider deleting it or replacing it with a description of exactly what's being zeroed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->

liangfok commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> since my remaining comments are all `BTW`.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-:-Ke0JvSoAh88kC4VSLv7:bozo4qp)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
+@david-german-tri here is another one for platform review. thanks! 

---

Review status: 6 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.cc, line 98 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdxG9ayuXXiv0ckvA_x:-Ke0LH2hGOF5-hXY9CnB:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.cc#L98)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider having this method call the method above.
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-Ke0L-1fdRQyKGyQoAPV:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, "everything related" is perhaps too informal and vague. Consider deleting it or replacing it with a description of exactly what's being zeroed.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-:-Ke0LiUdQw2L_R1cK6qY:b-o2epku)*
<!-- Sent from Reviewable.io -->

liangfok commented:
*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-Ke0MuMdu6Y5h0BkuWAN:bd9go0u) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

Done.
</blockquote></details>

BTW: typo: "meaningful".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 8 files at r1, 1 of 4 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-Ke0W066RhB7SkSIIGnn:bs6zpj1) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: typo: "meaningful".
</blockquote></details>

Rereading this after our f2f conversation - it's still not obvious to me from this comment that the torques will be zero-initialized, and consequently ignored.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 128 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Ke0N8JKGkKNkU2UvxUg:-Ke0N8JLriJStVz0cT99:bcluxjz) ([raw file](https://github.com/robotlocomotion/drake/blob/10b534f5bd86a4f3cb64412f9161d6aacf0aa8eb/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L128)):*
> ```objc
>    */
>   void UpdateKinematics(double t, const Eigen::Ref<const VectorX<double>>& q,
>               const Eigen::Ref<const VectorX<double>>& v);
> ```

BTW weird indent

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 16 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Ke0PSu3oc72peBLasZi:-Ke0PSu4nt2EuHKyPSTC:bti2zby) ([raw file](https://github.com/robotlocomotion/drake/blob/10b534f5bd86a4f3cb64412f9161d6aacf0aa8eb/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L16)):*
> ```objc
> 
> /**
>  * A base class for systems that translate various types of state into
> ```

BTW nit: "abstract base class"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-:-Ke0WABkA_fcqrUFIkXv:bnfp4nl)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 5 of 5 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/iiwa_wsg_simulation.cc, line 122 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Ke1brLrcgNuVGsHGM-m:-Ke1brLrcgNuVGsHGM-n:bldms2c) ([raw file](https://github.com/robotlocomotion/drake/blob/b191dffe08978a83bb472426a4a4e48656888b0c/drake/examples/kuka_iiwa_arm/iiwa_wsg_simulation.cc#L122)):*
> ```cpp
> 
>   auto plant = std::make_unique<RigidBodyPlant<T>>(tree_builder->Build());
>   // Contact parameters
> ```

Missing period at end of sentence.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
@drake-jenkins-bot linux-xenial-clang-experimental please

siyuanfeng-tri commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-Ke1YbsBLkcSH_5AaI62:bk7hcl8) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

Rereading this after our f2f conversation - it's still not obvious to me from this comment that the torques will be zero-initialized, and consequently ignored.
</blockquote></details>

i rephrased a bit. 

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 128 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Ke0N8JKGkKNkU2UvxUg:-Ke1YhM9BI_Ep_r4xmNa:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/10b534f5bd86a4f3cb64412f9161d6aacf0aa8eb/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L128)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW weird indent
</blockquote></details>

Done.

---

*[drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h, line 16 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-Ke0PSu3oc72peBLasZi:-Ke1Yqn9XkFDu8Z7nkpF:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/10b534f5bd86a4f3cb64412f9161d6aacf0aa8eb/drake/examples/QPInverseDynamicsForHumanoids/system/humanoid_status_translator_system.h#L16)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW nit: "abstract base class"
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-Ke4MeQ-Pg1BxPNnx4xN:b3qgb7o) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

i rephrased a bit. 
</blockquote></details>

BTW, no change?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-KdwiQvG-0IAqTJUNX2y:-Ke4Mn-F_56fbi_7lNFo:blu1is3) ([raw file](https://github.com/robotlocomotion/drake/blob/6c531c36ac49323cfc4c35debb25d08ee9a4f367/drake/examples/QPInverseDynamicsForHumanoids/humanoid_status.h#L123)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, no change?
</blockquote></details>

Sorry, ignore my comment. I see the change in `r4`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295#-:-Ke4qVwg7ZZ6XmuvJfcm:bnfp4nl)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 2 of 3 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5295)*
<!-- Sent from Reviewable.io -->
